pkg/cfn/manager/nodegroup.go

```diff
@@ -99,7 +104,39 @@ func (c *StackCollection) propagateManagedNodeGroupTagsToASGTask(ctx context.Con
 			asgNames = append(asgNames, *asg.Name)
 		}
 	}
-	return c.PropagateManagedNodeGroupTagsToASG(ng.Name, ng.Tags, asgNames, errorCh)
+
+	// add labels and taints
+	tags, err := convertLabelsAndTaintsIntoTags(ng)
+	if err != nil {
+		return err
+	}
+
+	// add nodegroup tags
+	for k, v := range ng.Tags {
+		tags[k] = v
+	}
+
+	return c.PropagateManagedNodeGroupTagsToASG(ng.Name, tags, asgNames, errorCh)
+}
+
+// convertLabelsAndTaintsIntoTags so that they can be propageted automatically to ASG afterwards
+func convertLabelsAndTaintsIntoTags(ng *api.ManagedNodeGroup) (map[string]string, error) {
```

Comment:
can we add tests for this?

Reply:
17ba36563440eb2e36085ead9ac7e8580b98f6a4